API/DTOs/LeaderboardFilterDTO.cs

```diff
@@ -1,52 +1,124 @@
+using System.ComponentModel.DataAnnotations;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
 namespace API.DTOs;
 
 /// <summary>
 /// Filters for the leaderboard
 /// </summary>
-public class LeaderboardFilterDTO
+public class LeaderboardFilterDTO : IValidatableObject
 {
     /// <summary>
-    /// The "better" inclusive bound (ranges from 1+)
+    /// Rank floor
     /// </summary>
-    public int? MinRank { get; set; }
+    /// <remarks>
+    /// The "better" inclusive rank bound.
+    /// If given, only players with a rank greater than or equal to this value will be included
+    /// </remarks>
+    [FromQuery]
+    [Range(1, int.MaxValue)]
+    public int? MinRank { get; init; }
 
     /// <summary>
-    /// The "worse" inclusive bound (ranges from 1+)
+    /// Rank ceiling
     /// </summary>
-    public int? MaxRank { get; set; }
+    /// <remarks>
+    /// The "worse" inclusive rank bound.
+    /// If given, only players with a rank less than or equal to this value will be included
+    /// </remarks>
+    [FromQuery]
+    [Range(1, int.MaxValue)]
+    public int? MaxRank { get; init; }
 
     /// <summary>
-    /// The lower-performing rating bound (ranges from 100+)
+    /// Rating floor
     /// </summary>
-    public int? MinRating { get; set; }
+    /// <remarks>
+    /// The "better" inclusive rating bound.
+    /// If given, only players with a rating greater than or equal to this value will be included
+    /// </remarks>
+    [FromQuery]
+    [Range(100, int.MaxValue)]
+    public int? MinRating { get; init; }
 
     /// <summary>
-    /// The higher-performing rating bound (ranges from 100+)
+    /// Rating ceiling
     /// </summary>
-    public int? MaxRating { get; set; }
+    /// <remarks>
+    /// The "worse" inclusive rating bound.
+    /// If given, only players with a rating less than or equal to this value will be included
+    /// </remarks>
+    [FromQuery]
+    [Range(100, int.MaxValue)]
+    public int? MaxRating { get; init; }
```

Comment:
MinRating and MaxRating remarks are backwards. The `MinRating` is worse, `MaxRating` is better.

Reply:
oh oops sick copy paste